app/Http/Filters/ContactCustomerFilter.php

```diff
@@ -79,9 +79,19 @@ public function filterBranch()
      * @param string $from
      * @param string $column
      */
-    public function startDate(string $from, $column=self::DATE)
+    public function startDate(string $from, $column = self::DATE)
     {
         $this->builder->whereDate($column, '>=', $from)
-            ->whereDate($column, '<=',$this->request->endDate ?? Carbon::now());
+            ->whereDate($column, '<=', $this->request->endDate ?? Carbon::now());
+    }
+
+    public function inActiveDays(int $days = 30)
+    {
+        // dd( Carbon::now()->subDays($days)->format('Y-m-d'));
+        $this->builder->whereHas('lastProspectActivity', function ($query) use ($days) {
+            $query->orderby('date', 'desc')
+                ->where('date', '<=', Carbon::now()->subDays($days)->format('Y-m-d'))
+                ->where('user_id', auth()->id());
```

Comment:
can you condition this based on access? check filterBranch for context

Reply:
The filter branch will be applied in the controller, the filterbyBranch on contact customer model does the job